src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -3333,5 +3333,57 @@ public static IEnumerable<object[]> CommentTestLineSeparators
                 };
             }
         }
+
+        public static IEnumerable<object[]> GetCommentTestData
+        {
+            get
+            {
+                var dataList = new List<object[]>();
+                foreach (var delim in new[] { "\r", "\r\n", "\n" })
+                {
+                    // NOTE: Leading and trailing spaces in the comments are significant.
+                    var singleLineComment = " Single Line Comment ";
+                    dataList.Add(new object[] { $"{{//{singleLineComment}{delim}}}", singleLineComment });
+
+                    var multilineComment = $" Multiline {delim} Comment ";
+                    dataList.Add(new object[] { $"{{/*{multilineComment}*/{delim}}}", multilineComment });
+                }
+                return dataList;
+            }
+        }
+
+        public static IEnumerable<object[]> GetCommentUnescapeData
+        {
+            get
+            {
+                var dataList = new List<object[]>();
+
+                var rawComments = new string[]
+                {
+                    "A string with {0}valid UTF8 \\t tab",
+                    "A string with {0}invalid UTF8 \\xc3\\x28",
+                    "A string with {0}valid UTF16 \\u002e \\u0009 म",
+                    "A string with {0}invalid UTF16 \\uDD1E"
+                };
+
+                // single line comments
+                foreach (var raw in rawComments)
+                {
+                    var str = string.Format(raw, "");
```

Comment:
Why do we need to do this?

Reply:
In here, each of the line in the `rawComments` works as format string. The `{0}` in each is replaced with `""` and generates a single line comment since it is prefixed with `//` in the loop.

Whereas, in the loop at 3378 same is replaced with `\n` and the resultant string when enclosed in `/*` and `*/` makes a new line comment...